zigup.zig

```diff
@@ -429,6 +448,44 @@ fn listCompilers(allocator: *Allocator) !void {
     }
 }
 
+fn keepOrUnkeepCompiler(allocator: *Allocator, compiler_name_to_keep_or_unkeep: []const u8, keep_or_unkeep: enum { keep, unkeep }) !void {
+    if (mem.eql(u8, "master", compiler_name_to_keep_or_unkeep)) {
+        std.debug.warn("master is always kept, doing nothing\n", .{});
+        return;
+    }
+    const install_dir_string = try getInstallDir(allocator, .{ .create = true });
+    defer allocator.free(install_dir_string);
+
+    // TODO openDirAbsolute in stdlib
+    var install_dir = std.fs.cwd().openDir(install_dir_string, .{ .iterate = true }) catch |e| switch (e) {
+        error.FileNotFound => return,
+        else => return e,
+    };
+    defer install_dir.close();
+
+    var compiler_dir = install_dir.openDir(compiler_name_to_keep_or_unkeep, .{}) catch |e| switch (e) {
+        error.FileNotFound => {
+            std.debug.warn("compiler not found: {}\n", .{compiler_name_to_keep_or_unkeep});
+            return;
+        },
+        else => return e,
+    };
+    if (keep_or_unkeep == .keep) {
+        var keep_fd = try compiler_dir.createFile("keep", .{});
+        keep_fd.close();
+        std.debug.warn("created '{}{c}{}{c}{}'\n", .{ install_dir_string, std.fs.path.sep, compiler_name_to_keep_or_unkeep, std.fs.path.sep, "keep" });
+    } else {
+        compiler_dir.deleteFile("keep") catch |e| switch (e) {
+            error.FileNotFound => {
+                std.debug.warn("compiler '{}' does not have keep file, so it can't be unkept\n", .{compiler_name_to_keep_or_unkeep});
```

Comment:
Not sure if this should be an error or not.  I would start out by making this an error.